### PR TITLE
[mdns] use `NextFireTime` class

### PR DESCRIPTION
**Pull request by abtink**
This commit updates the mDNS module to use the recently introduced general-purpose `NextFireTime` class, replacing the module-specific `TimerContext` class.

### CODE REVIEW

File: src/core/net/mdns.hpp [hunk 1]
```diff
@@ -1189,32 +1190,19 @@ class Core : public InstanceLocator, private NonCopyable
...
-    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-
-    class EntryTimerContext : public TimerContext // Used by `HandleEntryTimer`.
+    class EntryTimerContext : public InstanceLocator // Used by `HandleEntryTimer`.
     {
     public:
         EntryTimerContext(Instance &aInstance);
-        TxMessage &GetProbeMessage(void) { return mProbeMessage; }
-        TxMessage &GetResponseMessage(void) { return mResponseMessage; }
+        TimeMilli     GetNow(void) const { return mNextFireTime.GetNow(); }
```
Comment (Irving-cl) [inline src/core/net/mdns.hpp:1197]: It seems that the `GetNow` method of `EntryTimerContext` is not used in this PR. Is it required?

File: src/core/net/mdns.hpp [hunk 2]
```diff
@@ -1440,14 +1428,17 @@ class Core : public InstanceLocator, private NonCopyable
 
     // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-    class CacheTimerContext : public TimerContext
+    class CacheTimerContext : public InstanceLocator
     {
     public:
         CacheTimerContext(Instance &aInstance);
-        TxMessage &GetQueryMessage(void) { return mQueryMessage; }
+        TimeMilli     GetNow(void) const { return mNextFireTime.GetNow(); }
```
Comment (Irving-cl) [inline src/core/net/mdns.hpp:1435]: Same as above.

File: src/core/net/mdns.hpp (hunk 1 above)
Comment (abtink) [reply src/core/net/mdns.hpp:1197]: Yes, we definitely need the `GetNow()` method. It's used in many places throughout the existing code (unaffected by this PR), such as in `Core::Entry::HandleTimer(EntryTimerContext &aContext)`.  It was previously provided by the `TimerContext` base class.

File: src/core/net/mdns.hpp (hunk 2 above)
Comment (abtink) [reply src/core/net/mdns.hpp:1435]: Same here. For example used in `PrepareQuery(CacheTimerContext &aContext)`

File: src/core/net/mdns.hpp (hunk 1 above)
Comment (Irving-cl) [reply src/core/net/mdns.hpp:1197]: Got it. Thanks!

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**